### ISSUE TITLE
Timeout errors are not visible to the user

### ISSUE THREAD

**Issue by dennwc** [bblfsh/web#187]
### Steps to reproduce:
1) Go to http://play.bblf.sh
2) Change language to "JavaScript"
3) Paste a large file ([example](https://github.com/bblfsh/javascript-driver/blob/master/fixtures/bench_nodes.js))
4) Click "Parse"

### What happens
Nothing. there is still an old UAST in the right panel and no indication of failed parse request.

### What is expected
Looking at the console, there is clearly a timeout for parse request. It should be displayed to the user with a suggestion to try a smaller file.

**Comment by smacker** [comment]
it was partly fixed on master. Now it shows "internal error":
<img width="1110" alt="Screenshot 2019-03-12 at 12 53 43" src="https://user-images.githubusercontent.com/406916/54198097-ec95d800-44c5-11e9-81de-60bfe4f29df0.png">

**Comment by dpordomingo** [comment]
@smacker I think we should try to show that it was a timeout error, shouldn't we?
And even better if the dashboard can also say in which part of the communication did it happened

**Comment by smacker** [comment]
that's why I said "partially".

**Comment by smacker** [comment]
Current bblfshd and client-go don't have default timeouts anymore. We don't add any timeout too.
So this bug isn't reproducable locally anymore.

In a hosted enviroment proxy can timeout and this error is displayed correctly (I saw it but I can't reproduce it anymore for a screenshot, bblfshd became too fast).

So here is nothing to fix.

But I added correct handling of request cancelation & correct http error in case bblfsh will add some timeout in the future.
